cli/uninstall/uninstall.go

```diff
@@ -61,19 +61,30 @@ func remove(program string, programVersion string, directory string,
 	} else if err != nil {
 		return false, fmt.Errorf("there was some problem locating %s", path)
 	}
-	// Get path where symlink point.
-	resolvedPath, err := util.ResolveSymlink(linkPath)
-	if err != nil {
-		return false, fmt.Errorf("failed to resolve symlink %s: %s", linkPath, err)
-	}
+
 	var isSymlinkRemoved bool
-	// Remove symlink if it points to program.
-	if strings.Contains(resolvedPath, fileName) {
-		err = os.Remove(linkPath)
+
+	_, err = os.Lstat(linkPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return false, fmt.Errorf("failed to access %s: %s", linkPath, err)
+		}
+		isSymlinkRemoved = false
```

Comment:
Looks like, if there is no symlink, we report like it is not removed. Please add a test to check the message in case of missing symlink, but existing binary.